### PR TITLE
Changes & Adds Cryo Tube Sound & Alert Messages

### PR DESCRIPTION
**Pull request by JohnTheDonut**
## **Cryo Tube Sound**
Adds a new sound called cryo_warning.ogg this replaces the microwave ding alert sound with a much more "Robotic machine like" beeping sound effect. This should help medical staff more easily  distinguish if their donk pockets have finished being heated up in staff room microwave or if someone is trapped in a cryo tube.

## **New Cryo Tube Messages**
Adds new medical channel radio messages to the cryo tube if a patient has been fully restored the cryo tube will say "Patient fully restored" if auto eject is enabled then the cryo tube will also say "Auto ejecting patient now" This will help when a patient that  has been fully restored has accidentally been left in a cryo tube by a Medical Doctor as it will be a reminder for medical doctors on the station to eject the patient from the cryo tube.

## **Changelog**
:cl: Tofa01 & XDTM
Add: Adds radio alert messages going to medical channel to the cryo tube when a patient is fully restored.
Soundadd: Adds new alert sound for cryo tube. (cryo_warning.ogg)
/:cl:

### CODE REVIEW

**Comment by XDTM** [comment]
It should either send a radio message like cloning, or send local speech like the teleporter, visible message is used to describe things.

**Comment by Tofa01** [comment]
Hm, that's a good idea i am trying to reverse engineer how the cloning pod radio works right now.

**Comment by Tofa01** [comment]
Ok i seem to have it working but now when the cryo tube is opened a station bounced radio is there

**Comment by Tofa01** [comment]
Ok went through this with coderbus, long stroy short they are stumped so i am just going to make the messages Say("") messages so no radio messages.

**Comment by XDTM** [comment]
I made it work! The branch is here: https://github.com/XDTM/tgstation/tree/TofaCryo

**Comment by RandomMarine** [comment]
So, this just makes it more obvious when someone's swapped out the cryo mix with something deadly with an obvious OH NO PEOPLE ARE DYING message.

👎 Quality of life shouldn't take hinder sabotage/murder options.

**Comment by ExcessiveUseOfCobblestone** [comment]
>So, this just makes it obvious when someone's swapped out the cryo mix with something deadly with a blatant OH NO PEOPLE ARE DYING message and sound. And prevents trying to hide bodies in the cells in a pinch.

@Tofa01 does this affect sabotaging?

**Comment by Tofa01** [comment]
@ExcessiveUseOfCobblestone Indeed but is people think that is going too far i can either remove the message for if someone dies in the cryotube and just have it sit there in silence like before or i can make it a emag feature, what do you guys think?

**Comment by Tofa01** [comment]
@XDTM Very cool, thanks i was up until 3AM trying to get that to work and failed ill test it out now and commit it ASAP.

**Comment by PKPenguin321** [comment]
You need to learn the concept of things being intentionally imperfect. Things are intentionally not perfect to make room for conflicts, in this case sabotage.

:-1:

**Comment by Tofa01** [comment]
Thanks for the feedback, i think due to some obvious disagreements to the "Subject deceased message" ill remove it. @PKPenguin321 Yeah that makes sense, i am still pretty new and this is my first time adding a feature.

**Comment by Tofa01** [comment]
Radio is working properly now Thanks XDTM! Also removed "Subject Deceased" messages so no longer will the cryo tube interrupt "murdery sabotage"

**Comment by Tofa01** [comment]
Ok, this is ready to merge whenever if no one else has any issues.

File: code/modules/atmospherics/machinery/components/unary_devices/cryo.dm [hunk 1]
```diff
@@ -1,3 +1,5 @@
+#define SPEAK(message) radio.talk_into(src, message, radio_channel, get_spans())
```
Comment (Jordie0608) [inline code/modules/atmospherics/machinery/components/unary_devices/cryo.dm]: This really doesn't need to be a define.

Comment (Tofa01) [inline code/modules/atmospherics/machinery/components/unary_devices/cryo.dm]: That would be correct fixing this now.

Comment (Tofa01) [inline code/modules/atmospherics/machinery/components/unary_devices/cryo.dm]: @Jordie0608 Looks like it needs to be a define, when i try and use "radio.talk_into(src, "message here") it just broadcasts to the station wide radio

Comment (Tofa01) [inline code/modules/atmospherics/machinery/components/unary_devices/cryo.dm]: The radio.talk_into system is extremely picky about HOW and IF it wants to work.

Comment (Jordie0608) [inline code/modules/atmospherics/machinery/components/unary_devices/cryo.dm]: This just replaces the define with the value it's assigned during preprocessor, you would need to go the same; you've missed out the last two arguments.
Ideally the cloning pod also wouldn't do this as there's little value in defines like this.

**Comment by Tofa01** [comment]
Fixed! Hopefully we now _GO_ go launch.

**Comment by Tofa01** [comment]
Ready for merge.

**Comment by SweaterKittens** [comment]
Don't let these guys browbeat you, no one uses cryo for murder, and if a doctor is too dumb to check whether or not there's anyone in the cryotube, what's in the beakers, or if their patient is dying, a message stating that the patient has died isn't going to fix that. 

This will be a really nice change for when things get hectic in medbay.

**Comment by Tofa01** [comment]
Indeed but the issue is that when i receive messages from two people in a row in very short time frames i thought that it may be what the popular public opinion could be though now looking at it there is now way to tell whether this it is simply the vocal minority or the public majority unless this PR gets more people looking at it and more feedback etc.

**Comment by XDTM** [comment]
Not sending a message on death is a good change, since tube murder is rare but it can happen, and it's original enough that it shouldn't be discouraged. I like the rest of course.

**Comment by Tofa01** [comment]
Thanks @XDTM I agree i am pretty happy with the end result of this PR, thanks for the help with the radio code i added your name to the changelog. At this point i am just waiting for review/merge.

**Comment by jughu** [comment]
@tofa1 Smell, if it's cryo they smell the burning of the patient before they go in - with the donk pockets the burning happens after. Thats how they figure out the difference

**Pull request merged**